packages/netlify-cms-lib-util/src/implementation.ts

```diff
@@ -183,9 +183,7 @@ const fetchFiles = async (
           try {
             const [data, fileMetadata] = await Promise.all([
               readFile(file.path, file.id, { parseText: true }),
-              file.id
```

Comment:
File collections entries don't have an id (just a file path), so `readFileMetadata` was not called for them. This fixes it